### PR TITLE
Update license from bsd-3 to apache-2

### LINKED ISSUE #29: Add project license
Need to add opensource license

suggesting Apache 2.0

### PR DESCRIPTION
**Pull request by dlstadther**
# Motivation

Update license to Apache 2 to allow broader use (including private/patent/trademark usage).

# Description

* Swap pyproject license reference (updates pypi license)
* Add `LICENSE` to repo
* Add license badge

Closes #29 

# Testing

Only able to test the render of the license badge. Links et al are dependent on merge to upstream.

### CODE REVIEW

**Review by gage-russell — DISMISSED** [review]
Awesome dude, thank you!!

If any issues rendering anything after read, feel free to just open a quick fix!

**Comment by gage-russell** [comment]
https://github.com/gage-russell/pandas-lineage/pull/30 has been merged! will re-approve here after conflict resolution!

File: LICENSE
Comment (gage-russell) [inline LICENSE:189]: Should this be filled in? (honest question although it might sound silly)

Comment (dlstadther) [reply LICENSE:189]: It should. Good catch.

Comment (dlstadther) [reply LICENSE:189]: Actually no. This is a message about what the boilerplate notice should be in each file.

e.g.
[Luigi](https://github.com/spotify/luigi/blob/master/luigi/__main__.py)
[Marquez](https://github.com/MarquezProject/marquez/blob/main/api/src/main/java/marquez/api/BaseResource.java)


For the time-being, i'm just going to add the license file to the repo and address the inclusion of boilerplate to code files later.

**Review by gage-russell — APPROVED** [review]
Thank you!!!

**Comment by gage-russell** [comment]
Plan on merging this tomorrow (Sunday). Curious though.. are you able to merge after being approved? / should you be able to based on experience in past open source projects?

**Comment by dlstadther** [comment]
@gage-russell , no rush from me!

I'm not able to merge. You'd have to grant that permission to me as a repo collaborator, i think.

For opensource projects, you typically see a select few who are permitted to merge PRs - these are your "maintainers". Beyond that, contributors can create forks, new branches, and submit their feature branch's upstream (to your repo) as PRs. Then it's up to the maintainers and community to review, and maintainers to merge (if approved).

**Pull request merged**
